### ISSUE TITLE
Failing "-x" command line option

### ISSUE THREAD

**Issue by donv** [jruby/jruby#4536]
Given a script `mixed_script`:

```ruby
#!/bin/sh
echo 'bash part'
'exec' "ruby" '-x' "$0" "$@"
#!ruby
puts 'ruby part'
```

Run with

```
./mixed_script
```

With MRI installed as current Ruby this produces

```
bash part
ruby part
```

With JRuby this produces

```
bash part
jruby: no Ruby script found in input (LoadError)
```

### Environment

Darwin Gratass.local 16.4.0 Darwin Kernel Version 16.4.0: Thu Dec 22 22:53:21 PST 2016; root:xnu-3789.41.3~3/RELEASE_X86_64 x86_64
jruby 9.1.8.0 (2.3.1) 2017-03-06 90fc7ab Java HotSpot(TM) 64-Bit Server VM 25.112-b16 on 1.8.0_112-b16 +jit [darwin-x86_64]
jruby 9.1.5.0 (2.3.1) 2016-09-07 036ce39 Java HotSpot(TM) 64-Bit Server VM 25.112-b16 on 1.8.0_112-b16 +jit [darwin-x86_64]
jruby 1.7.26 (1.9.3p551) 2016-08-26 69763b8 on Java HotSpot(TM) 64-Bit Server VM 1.8.0_112-b16 +jit [darwin-x86_64]

**Comment by donv** [comment]
I could use some help with this.  Is is preventing my next deployment.

I can find where the error message comes from:

https://github.com/jruby/jruby/blob/b3c3613df6affdb8db1b891e60ad48c5cd2c5ced/core/src/main/java/org/jruby/Main.java#L299

This is based on `RubyInstanceConfig#hasShebangLine` which is set here:

https://github.com/jruby/jruby/blob/b1e84356a81097da873e437201f9bcf9540bd301/core/src/main/java/org/jruby/RubyInstanceConfig.java#L284

That is in `RubyInstanceConfig#parseShebangOptions`, but I cannot find where this is called.  Maybe nowhere?

**Comment by kschoelhorn** [comment]
I also just stumbled upon this (exactly two years later :grin:).

The option probably broke with the fix for #3294: In the first pass `doProcessArguments`, `parseShebangOptions`, and finally `setHasShebangLine` was called, so `hasShebangLine` returned true afterwards. There is event a [test](https://github.com/jruby/jruby/blob/d54c5d0c5464bfd379676f4cf3916ee605f56bce/spec/ruby/command_line/dash_x_spec.rb) for it, but it seems it is not run?

I have a fix in the making, will open a PR if it is finished.
